src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -70,21 +69,19 @@ export const ColorPicker: FC<ColorPickerProps> = ({
 };
 
 const ColorPreview: FC<{ color: Color; format: ColorFormat }> = ({ color, format }) => {
-    const { hex, rgba, name, alpha } = color;
-    const backgroundColor = getBackgroundColor(color);
+    const parsedColor = tinycolor(color);
+    const backgroundColor = parsedColor.toRgbString();
     const displayValue = getColorDisplayValue(color, format);
-    const labelColor = useMemo(() => {
-        return alpha && alpha < 0.3 ? null : getContrastingColor(hex);
-    }, [hex, rgba, alpha]);
-
     return (
         <div
-            className="tw-flex tw-flex-none tw-justify-center tw-p-7 tw-text-m tw-text-black dark:tw-text-white tw-gap-2"
-            style={{ background: backgroundColor, color: labelColor }}
+            className={`tw-flex tw-flex-none tw-justify-center tw-p-7 tw-text-m tw-gap-2 ${
+                isColorLight(color) ? "tw-text-black" : "tw-text-white"
+            }`}
+            style={{ background: backgroundColor }}
             data-test-id="color-preview"
         >
-            {name && <span className="tw-font-bold">{name}</span>}
-            <span className={name ? "" : "tw-font-bold"}>{displayValue}</span>
+            {color.name && <span className="tw-font-bold">{color.name}</span>}
```

Comment:
You could create a `const colorName = color.name` above and reuse it

Reply:
Added